src/utils/fetchClient.ts

```diff
@@ -29,7 +29,13 @@ function request<T>(
   // for a demo purpose we emulate a delay to see if Loaders work
   return wait(300)
     .then(() => fetch(BASE_URL + url, options))
-    .then(response => response.json());
+    .then(response => {
+      if (!response.ok) {
+        throw new Error();
```

Comment:
Throwing a generic error here provides no context about what went wrong. Consider including the response status or message in the error to make debugging easier and provide more informative feedback to the user.